### PR TITLE
[BIOMAGE-1860] - Add rotating secrets to rds

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-1860

#### Link to staging deployment URL 
N/A

#### Links to any Pull Requests related to this
N/A

#### Anything else the reviewers should know about the changes here
- Added rotation for the secret
- Added resource-based policy for the secret

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock  (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/hms-dbmi-cellenics/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/hms-dbmi-cellenics/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/hms-dbmi-cellenics/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: cf/rds.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 AWSTemplateFormatVersion: "2010-09-09"
+Transform: AWS::SecretsManager-2020-07-23
```
Comment (cosa65) [inline cf/rds.yaml:4]: What does this `Transform` do?

File: cf/rds.yaml [hunk 2]
```diff
@@ -35,6 +36,40 @@ Parameters:
...
+      SecretId: !Ref RDSMasterUserSecret
+      HostedRotationLambda:
+        RotationType: PostgreSQLSingleUser
+        RotationLambdaName: !Sub "RDSSecretsManagerRotation${Environment}"
+        VpcSubnetIds:
+          Fn::ImportValue: !Sub "eksctl-biomage-${Environment}-cluster::SubnetsPrivate"
+        VpcSecurityGroupIds:
+          Fn::ImportValue: !Sub "eksctl-biomage-${Environment}-cluster::ClusterSecurityGroupId"
+      RotationRules:
+        Duration: 2h
```
Comment (cosa65) [inline cf/rds.yaml:78]: This rotates every 2 hours? Is it necessary to do it so frequently I looked at some docs on rotation and usually they do it every 30 days or something

**Review by cosa65 — COMMENTED** [review]
We need to add the `Deny` ruel mentioned in the ticket too so that only the aws admins can access teh secret

File: cf/rds.yaml (hunk 2 above)
Comment (aerlaut) [reply cf/rds.yaml:78]: No, that is the rotation time window. That means the key will be rotated within 2 hours of the schedule.

Comment (cosa65) [reply cf/rds.yaml:78]: ah ok, then how frequent is the rotation?

Comment (aerlaut) [reply cf/rds.yaml:78]: The rotation is done once every calendar months, on the 1st day between 8AM - 10 AM.

Comment (aerlaut) [reply cf/rds.yaml:78]: This is defined in `ScheduleExpression: 'cron(0 8 1 * ? *)`

File: cf/rds.yaml (hunk 1 above)
Comment (aerlaut) [reply cf/rds.yaml:4]: `Transform` allows AWS to inject CF resources on our behalf. [In our case](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/transform-aws-secretsmanager.html), this is required to use the AWS-provided [secret rotation lambda](https://docs.aws.amazon.com/secretsmanager/latest/userguide/reference_available-rotation-templates.html#sar-template-postgre-singleuser).

File: cf/rds.yaml [hunk 3]
```diff
@@ -51,13 +94,12 @@ Resources:
...
-      MasterUsername: !Join ["", ["{{resolve:secretsmanager:", !Join ['-', [!Ref DBSecretPrefix, !Ref Environment]], ":SecretString:username}}" ]]
-      MasterUserPassword: !Join ["", ["{{resolve:secretsmanager:", !Join ['-', [!Ref DBSecretPrefix, !Ref Environment]], ":SecretString:password}}" ]]
+      MasterUsername: !Sub "{{resolve:secretsmanager:${RDSMasterUserSecret}::username}}"
+      MasterUserPassword: !Sub "{{resolve:secretsmanager:${RDSMasterUserSecret}::password}}"
       EnableIAMDatabaseAuthentication: true
       BackupRetentionPeriod: 5
       SourceRegion: !Sub '${AWS::Region}'
-      # Check if we want to change this
-      StorageEncrypted: false
+      StorageEncrypted: true
```
Comment (aerlaut) [inline cf/rds.yaml:102]: Encryption is set to true

File: cf/rds.yaml (hunk 2 above)
Comment (cosa65) [reply cf/rds.yaml:78]: Nice, coukld you add a comment next to `ScheduleExpression` saying that the rotation is every 1 month?

**Review by cosa65 — APPROVED** [review]

**Pull request merged**